### ISSUE TITLE
Ability to show system-out

### ISSUE THREAD

**Issue by marcdacz** [lukejpreston/xunit-viewer#46]
Firstly, thanks for making this module. It's awesome!
I'm using it in my test framework myself.

Anyway, I tried looking for this ability in xunit-viewer and it might just be a mistake on my end, but I am unable to show **_system-out_** fields from JUnit.xml 
Could you please confirm if this is supported and if not, are you planning to support it?

Thanks heaps!

-marcDacz

**Comment by lukejpreston** [comment]
It doesn't sound supported, but could you give an example XML with system-out fields

**Comment by marcdacz** [comment]
Thanks for the quick response @lukejpreston, really appreciate it.
Here is a sample xml with a system-out:
Gist: https://gist.github.com/marcdacz/ec6f9b3da6b5316e3300bfc24943c5c8

**Comment by lukejpreston** [comment]
I haven't come across `system-out` before but I can make xunit-viewer show it. If you have control over this I would suggest you put that contents in the body of the xml e.g.

At least you will be able to see the message until I get around to doing the work (recently found a bunch of time to do this so expect radical changes!)

```xml
<testcase classname="scenario.test" name="Using Default Method and Endpoint" time="1.20s">
<![CDATA[{
  "requestBody": {
    "id": 12345,
    "title": "Using default method and endpoint",
    "body": "This test uses the default method and endpoint"
  },
  "responseData": {
    "id": 101,
    "title": "Using default method and endpoint",
    "body": "This test uses the default method and endpoint"
  },
  "responseStatus": 201
}]]>
</testcase>
```

**Comment by marcdacz** [comment]
Awesome @lukejpreston! Thanks for the help.
I actually don't have control over it as I am using a module called junit-report-builder.

Can't seem to see anyway to add it under the testcase (I could be wrong): https://github.com/davidparsson/junit-report-builder/blob/master/src/test_case.js

I'm willing to wait anyway 👍 

Thanks heaps!

**Comment by EltonZhong** [comment]
> Awesome @lukejpreston! Thanks for the help.
> I actually don't have control over it as I am using a module called junit-report-builder.
> 
> Can't seem to see anyway to add it under the testcase (I could be wrong): https://github.com/davidparsson/junit-report-builder/blob/master/src/test_case.js
> 
> I'm willing to wait anyway 👍
> 
> Thanks heaps!

you can just do something like `sed  s/<system-out>//g result.xml`, remove the xml tags and the output will be shown.

**Comment by mike-d-davydov** [comment]
@lukejpreston , system-out seem to be a standard jUnit.xml field, see for instance, https://llg.cubic.org/docs/junit/ . We need this feature as well. Please advise if contributions are welcome - I can give it a try. Thanks!

**Comment by lukejpreston** [comment]
I think this is working in 6.0.1

If someone could test it out and let me know that would be appreciated

If not then please share XML sample and I will get it working

**Comment by duckinator** [comment]
@lukejpreston it shows the output, but marks it as `?` (which I assume means unknown if it passed or failed) even though it passed.
![Screenshot_20200122_102532](https://user-images.githubusercontent.com/39698/72908513-90196100-3d03-11ea-8295-7b66284e13ef.png)

**Comment by lukejpreston** [comment]
Could you give a sample XML file then I can see what is going on?

Usually `?` means the test has a status value but it isn't a known one, if it had no status then it is `passed` by default

**Comment by duckinator** [comment]
@lukejpreston here you go: https://pastes.boop.ws/xunit.xml

**Comment by lukejpreston** [comment]
Thanks for that! It was really useful

Try upgrading to `6.0.13` which will have this fix.

I think I will close this issue but if anyone else notices anything strange around system-out let me know

**Comment by duckinator** [comment]
Yup, that works! Thanks!

**Comment by codebreach** [comment]
Bazel outputs system-out to testsuite and not the testcase element. Can we extend to support that as well?

ex:
```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="XYZ/e2e_tests" tests="1" failures="0" errors="1">
    <testcase name="XYZ/e2e_tests" status="run" duration="9" time="9">
      <error message="exited with error code 1"></error>
    </testcase>
    <system-out>
Generated test.log (if the file is not UTF-8, then this may be unreadable):
      <![CDATA[exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //XYZ
-----------------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/text_models/bazel/sandbox/linux-sandbox/8/execroot/text_models/bazel-out/k8-fastbuild/bin/XYZ/e2e_tests.runfiles/text_models/XYZ/e2e_tests.py", line 6, in <module>
    from XYZ.extractor import Extractor
  File "/tmp/text_models/bazel/sandbox/linux-sandbox/8/execroot/text_models/bazel-out/k8-fastbuild/bin/XYZ/e2e_tests.runfiles/text_models/XYZ/extractor.py", line 7, in <module>
    from ABC.DEF.helper import join_set
ModuleNotFoundError: No module named 'ABC']]>
    </system-out>
  </testsuite>
</testsuites>
```

**Comment by lukejpreston** [comment]
thanks for the example @codebreach I will look into this

**Comment by lukejpreston** [comment]
@codebreach I have added support to show the system out for a suite, this is available in `6.0.14` if you wouldn't mind testing it and confirm it resolves the issue that would be greatly appreciated :D

**Comment by codebreach** [comment]
Can confirm that this fixed it.

Starting from `6.0.14` I am able to run `xunit-viewer -r bazel-testlogs/SOME_PATH/ -s` to see a nicely formatted view of bazel results.
